io-engine/src/subsys/nvmf/subsystem.rs

```diff
@@ -165,6 +196,189 @@ fn make_sn<T: AsRef<[u8]>>(uuid: T) -> CString {
 }
 
 impl NvmfSubsystem {
+    pub fn extract_nexus_name(nqn: &str) -> Option<String> {
+        let vec: Vec<String> = nqn.split(':').map(|s| s.to_string()).collect();
+
+        if vec.len() != 2 {
+            return None;
+        }
+        Some(vec[1].clone())
+    }
+
+    fn reset_cb(success: bool, ctx: *mut c_void) {
+        if success {
+            info!(
+                "nvmf_event_handler: controller successfully reset {:?}",
+                success
+            );
+        } else {
+            error!("nvmf_event_handler: failed to reset controller ");
+        }
+        done_cb(ctx, success);
+    }
+
+    pub async fn reset_one_child(dev_name: String, ctx: *mut c_void) {
```

Comment:
This function is marked as 'async' but it doesn't use any '.await', so it's synchronous.
I'd suggest to turn it into fully asynchronous by implementing asynchronous awaiting of the operation inside this function. So we could move this code from 'reset_nexus()':

```
let (sender, receiver) = oneshot::channel::<bool>();
```

to 'reset_one_child()' and have the waiting logic hidden from the callers also in 'reset_one_child()':

```
if !receiver.await.expect("Reset completed") {
```

So the callers just do '::reset_one_child().await'.

Reply:
I tried this way first, but it didn't work. It got stuck at reset_cb->done_cb

Reply:
IMHO, it should work. It's the same reactor -  Master core. If it doesn't, it worth  investigating, IMHO.

Reply:
Looks like I had a different issue when I tried last time, '::reset_one_child().await works. Thanks for the suggestion.